Test Failure: DocumentSegment position assertion fails due to unordered database query

### Self Checks

- [x] I have read the [Contributing Guide](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md) and [Language Policy](https://github.com/langgenius/dify/issues/1542).
- [x] This is only for refactoring, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [x] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [x] I confirm that I am using English to submit this report, otherwise it will be closed.
- [x] 【中文用户 & Non English User】请使用英语提交，否则会被关闭 ：）
- [x] Please do not modify this template :) and fill in all the required fields.

### Description

The integration test `test_batch_create_segment_to_index_task_success_text_model` is failing with the following error:
```bash
FAILED api/tests/test_containers_integration_tests/tasks/test_batch_create_segment_to_index_task.py::TestBatchCreateSegmentToIndexTask::test_batch_create_segment_to_index_task_success_text_model - assert 3 == (0 + 1)
 +  where 3 = <models.dataset.DocumentSegment object at 0x7fe4f6ff7250>.position
```

### Motivation

_No response_

### Additional Context

_No response_

--- a/api/tests/test_containers_integration_tests/tasks/test_batch_create_segment_to_index_task.py
+++ b/api/tests/test_containers_integration_tests/tasks/test_batch_create_segment_to_index_task.py
@@ -296,7 +296,12 @@ class TestBatchCreateSegmentToIndexTask:
         from extensions.ext_database import db
 
         # Check that segments were created
-        segments = db.session.query(DocumentSegment).filter_by(document_id=document.id).all()
+        segments = (
+            db.session.query(DocumentSegment)
+            .filter_by(document_id=document.id)
+            .order_by(DocumentSegment.position)
+            .all()
+        )
         assert len(segments) == 3
 
         # Verify segment content and metadata

